src/DataValues/DecimalValue.php

```diff
@@ -355,10 +355,11 @@ public function getArrayValue() {
 	 *
 	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
 	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
-	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
+	 *  newFromArrayValue). Instead, use the constructor or use DataValue builder
+	 *  callbacks. (e.g. @see DataValueDeserializer)
```

Comment:
> […] move the full stop to the end […]

Sure, that's what I'm asking for. Why do I get a "come on" for requesting this?

> assuming one must use some static constructor in those deserializer callbacks.

I don't get where I'm saying this. What I say is "Instead, use the default constructor **or** […]". There is an or. In most cases the default constructor will be enough.

> saying "introduce a new X […] in the deprecation notice about removing "X".

I don't get where I say this either. Again: a static constructor might be convenient, but it should not be called "newFromArray". Reasons for this are explained extensively in the posts above.